carousel/index.html

```diff
@@ -0,0 +1,976 @@
+        // "touchmove" event to scroll the div with the user's X touch
+        container.addEventListener(
+          "touchmove",
+          function (e) {
+            if (iamInfo.carousel.isDragging) {
+              if (e.isCancelable) {
+                e.preventDefault();
+              }
+              var touchOffsets = e.changedTouches[0];
+              var diffX = iamInfo.carousel.dragStartX - touchOffsets.pageX;
+              iamInfo.carousel.dragStartX = touchOffsets.pageX;
+
+              var target = iamInfo.carousel.totalScroll + diffX;
+              applyScroll(target, true);
+
+              // decide if we have dragged far enough to prevent an clicking
+              var dragDistance = Math.abs(
+                iamInfo.carousel.dragStartX - iamInfo.carousel.rawDragStart
+              );
+              if (dragDistance > CAROUSEL_CONFIG.CLICK_DRAG_THRESHOLD) {
+                iamInfo.allowClicks = false;
+              }
+            }
+          },
+          false
+        );
+
+        container.addEventListener(
+          "mousemove",
+          function (e) {
+            if (iamInfo.carousel.isDragging) {
+              if (e.isCancelable) {
+                e.preventDefault();
+              }
+              var diffX = iamInfo.carousel.dragStartX - e.clientX;
+              iamInfo.carousel.dragStartX = e.clientX;
+
+              var target = iamInfo.carousel.totalScroll + diffX;
+              applyScroll(target, true);
+
+              // decide if we have dragged far enough to prevent an clicking
+              var dragDistance = Math.abs(
+                iamInfo.carousel.dragStartX - iamInfo.carousel.rawDragStart
+              );
+              if (dragDistance > CAROUSEL_CONFIG.CLICK_DRAG_THRESHOLD) {
+                iamInfo.allowClicks = false;
+              }
+            }
+          },
+          false
+        );
+
+        // "touchend" event to snap to prev, current, or next page
+        container.addEventListener(
+          "touchend",
+          function (e) {
+            if (iamInfo.carousel.isDragging) {
+              if (e.isCancelable) {
+                e.preventDefault();
+              }
+
+              handleEndOfDrag();
+            }
+          },
+          false
+        );
+
+        container.addEventListener(
+          "mouseup",
+          function (e) {
+            if (iamInfo.carousel.isDragging) {
+              if (e.isCancelable) {
+                e.preventDefault();
+              }
+
+              handleEndOfDrag();
+            }
+          },
+          false
+        );
+
+        container.addEventListener(
+          "mouseleave",
+          function (e) {
+            if (iamInfo.carousel.isDragging) {
+              if (e.isCancelable) {
+                e.preventDefault();
+              }
+
+              handleEndOfDrag();
+            }
+          },
+          false
+        );
+      });
+
+      // NOTE: for "resize" and "orientationchange" events
```

Comment:
i like that this comment explains why we're doing the multiple calls to setupCarouselPages in timeouts, but i'm wondering how relevant this is for our intended audience (non-technical folks)? 

last sentence in this comment needs revising:
> we are opting to make multiple calls function calls.